sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/OutputWriter.scala

```diff
@@ -57,6 +57,9 @@ abstract class OutputWriterFactory extends Serializable {
  * executor side.  This instance is used to persist rows to this single output file.
  */
 abstract class OutputWriter {
+  /** Initializes before writing any rows. Invoked on executor size. */
+  def init(): Unit = {}
```

Comment:
I don't think we need to add init only because one case CSV needs it.

Reply:
do the init logic in the constructor for CsvOutputWriter instead?